### ISSUE TITLE
Add support for Lambda@Edge

### ISSUE THREAD

**Issue by LegNeato** [srijs/rust-aws-lambda#28]
Because Lambda@Edge supports [Node 8.10](https://aws.amazon.com/about-aws/whats-new/2018/05/lambda-at-edge-adds-support-for-node-js-v8-10/), it has wasm support. This means we can run Rust there!

* [ ] Add support for [CloudFront events](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-event-structure.html)
* [ ] Add an easy way to define handler function / runtime using [`wasm-bindgen`](https://github.com/rustwasm/wasm-bindgen)

Open questions:

* Does it makes sense to be in this crate?
* Should we keep the same "runtime" api and just switch behavior with `cfg(target_arch = "wasm32")`?

**Comment by srijs** [comment]
I love the idea of just being able to use the `aws_lambda` crate and swap in a different backend to run on Node.js!

Whether that should go into `aws_lambda_runtime` or not probably depends a lot on whether we can make the existing `Runtime` API work for Node.js/wasm. I'll start to take a look at this!

**Comment by LegNeato** [comment]
Yeah, I was pretty excited too! I did some experiments today and I *think* we can make the current API work. Lambda@Edge basically just calls a JS function with a passed in callback, and the possible event types are static and spelled out....and then it is just generalizing from there for the to get to the NodeJS case. `wasm-bindgen` supports throwing JS exceptions so I think we are covered on the backtrace front (or at least it handles `Result`). The context api plumbing will have to change as well but I see no reason why the user-facing API does.

I got the runtime to build for wasm, it just needed some deps moved to target-specific sections and code annotations of course.
